packages/headless/src/controllers/core/result-list/headless-core-result-list.ts

```diff
@@ -50,7 +50,7 @@ export interface ResultListProps {
    * The action creator to build the `fetchMoreResults` action.
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  fetchMoreResultsActionCreator?: () => AsyncThunkAction<unknown, void, any>;
+  fetchMoreResultsActionCreator: () => AsyncThunkAction<unknown, void, any>;
```

Comment:
I'm making this (and by extension, the controller's `props` parameter) mandatory: we should always pass a fetchMoreResultsActionCreator when building this controller, otherwise the `fetchMoreResults` method won't do anything.

Reply:
I ended up rollings this back; the ResultListProps interface is used in atomic too so it broke stuff.